File: LeetCode/1975-Maximum-Matrix-Sum.cpp
// 1975. Maximum Matrix Sum

class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        long long sum = 0;
        int cnt = 0, mn = INT_MAX;
        int n = matrix.size(), m = matrix[0].size();
        bool has_zero = false;
        for(int i = 0; i < n; i++) {
            for(int j = 0; j < m; j++) {
                sum += abs(matrix[i][j]);
                mn = min(mn, abs(matrix[i][j]));
                if(matrix[i][j] < 0) {
                    cnt++;
                }
                else if(matrix[i][j] == 0) {
                    has_zero = true;
                }
            }
        }
        if(!has_zero && (cnt & 1)) {
            sum -= 2 * mn;
        }
        return sum;
    }
};

File: LeetCode/1975-Maximum-Matrix-Sum.java
// 1975. Maximum Matrix Sum

class Solution {
    public long maxMatrixSum(int[][] matrix) {
        int cntNegative = 0, mn = Integer.MAX_VALUE;
        long sum = 0;
        for (int[] row : matrix) {
            for (int cell : row) {
                if (cell < 0) {
                    cntNegative++;
                }
                mn = Math.min(mn, Math.abs(cell));
                sum += Math.abs(cell);
            }
        }
        if (cntNegative % 2 == 1) sum -= 2 * mn;
        return sum;
    }
}
